### DISCUSSION TITLE
Delay extraction

### DISCUSSION THREAD

**Discussion by salmen20** [The-OpenROAD-Project/OpenROAD-flow-scripts#2594] [Q&A]
Hello everyone,
after running the whole flow of my desing i want to extract some information practically  my design will comport two big module which will be instantiated in many places i want to know the delat of each of them .
Is there any simple way to do that ? 
Thanks .

**Comment by maliberty** [comment]
Is delat meant to be delay?  If so, what does it mean to know the delay of a module?

**Reply by salmen20** [reply]
yes i meant delay , by the delay of a module in my case i have two big modules both of them will mainly be composed by Multiplexers what i want to know is the time needed by the signal to navigate through that module

**Reply by maliberty** [reply]
There are many paths so I guess you are looking for some max delay.  `report_checks -through <inputs> -through <outputs>` would give you the worst timing path.  You could use `write_timing_model` manually inspect the model.

**Reply by salmen20** [reply]
Thank you , but will doing that require the GUI opened ? as the designs are big and will take more processing power than i have , is there a way to run those commands and having the reports without opening the GUI ? and Thanks

**Reply by maliberty** [reply]
No, you can report paths in text mode

**Comment by salmen20** [comment]
Yes i meant where can i run that command ? In the TCL part in the gui or
theres another way

**Reply by maliberty** [reply]
Just start openroad without the gui and use the CLI
